packages/cardano-services/src/Asset/CardanoTokenRegistry.ts

```diff
@@ -0,0 +1,104 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import { TokenMetadataService } from './types';
+import axios, { AxiosInstance } from 'axios';
+
+const DEFAULT_METADATA_SERVER_URI = 'https://tokens.cardano.org';
+
+/**
+ * Configuration options for CardanoTokenRegistry
+ */
+export interface CardanoTokenRegistryConfiguration {
+  metadataServerUri?: string;
+}
+
+interface CardanoTokenRegistryConfigurationWithRequired extends CardanoTokenRegistryConfiguration {
+  metadataServerUri: string;
+}
+
+/**
+ * Dependencies that are need to create CardanoTokenRegistry
+ */
+export interface CardanoTokenRegistryDependencies {
+  logger: Logger;
+}
+
+/**
+ * TokenMetadataService implementation using Cardano Token Registry public API
+ */
+export class CardanoTokenRegistry implements TokenMetadataService {
+  #axiosClient: AxiosInstance;
+  #cache: Record<string, Asset.TokenMetadata | null> = {};
```

Comment:
You can use the [InMemoryCache](https://github.com/input-output-hk/cardano-js-sdk/blob/feature/ADP-1901-Ogmios-time-settings/packages/cardano-services/src/InMemoryCache/InMemoryCache.ts) (this branch includes a refactor to make it generic compared to what's in _master_)